packages/account/src/Sections/Verification/ProofOfAddress/proof-of-address-form.tsx

```diff
@@ -0,0 +1,415 @@
+
+let file_uploader_ref: TFileRef;
+
+type TProofOfAddressForm = {
+    is_resubmit: boolean;
+    is_qualified_for_cfd_modal?: boolean;
+    onCancel: () => void;
+    onSubmit: (needs_poi: boolean) => void;
+    onSubmitForCFDModal: (index: number, values: FormikValues) => void;
+    step_index: number;
+};
+
+type TFormInitialValues = Record<
+    'address_line_1' | 'address_line_2' | 'address_city' | 'address_state' | 'address_postcode',
+    string
+>;
+
+type TFormState = Record<'is_btn_loading' | 'is_submit_success' | 'should_allow_submit' | 'should_show_form', boolean>;
+
+const ProofOfAddressForm = observer(
+    ({
+        is_resubmit,
+        is_qualified_for_cfd_modal,
+        onCancel,
+        onSubmit,
+        onSubmitForCFDModal,
+        step_index,
+    }: Partial<TProofOfAddressForm>) => {
+        const { client, notifications } = useStore();
+        const { account_settings, fetchResidenceList, fetchStatesList, getChangeableFields, is_eu, states_list } =
+            client;
+        const {
+            addNotificationMessageByKey: addNotificationByKey,
+            removeNotificationMessage,
+            removeNotificationByKey,
+        } = notifications;
+        const [document_file, setDocumentFile] = React.useState({ files: [], error_message: null });
+        const [is_loading, setIsLoading] = React.useState(true);
+        const [form_values, setFormValues] = React.useState<TFormInitialValues>({
+            address_line_1: '',
+            address_line_2: '',
+            address_city: '',
+            address_state: '',
+            address_postcode: '',
+        });
+        const [api_initial_load_error, setAPIInitialLoadError] = React.useState(null);
+        const [form_state, setFormState] = React.useState<TFormState>({
+            is_btn_loading: false,
+            is_submit_success: false,
+            should_allow_submit: true,
+            should_show_form: true,
+        });
+
+        React.useEffect(() => {
+            fetchResidenceList?.().then(() => {
+                Promise.all([fetchStatesList(), WS.wait('get_settings')]).then(() => {
+                    setFormValues({
+                        address_line_1: account_settings.address_line_1 ?? '',
+                        address_line_2: account_settings.address_line_2 ?? '',
+                        address_city: account_settings.address_city ?? '',
+                        address_state: account_settings.address_state ?? '',
+                        address_postcode: account_settings.address_postcode ?? '',
+                    });
+                    setIsLoading(false);
+                });
+            });
+        }, [account_settings, fetchResidenceList, fetchStatesList, is_eu]);
+
+        const changeable_fields = [...getChangeableFields()];
+
+        const validateFields = (values: TFormInitialValues) => {
+            (Object.entries(values) as ObjectEntries<TFormInitialValues>).forEach(
```

Comment:
```suggestion
            Object.entries(values).forEach(
```
@yauheni-deriv should work without assertion.  :)